### DISCUSSION TITLE
Ios build failing

### DISCUSSION THREAD

**Discussion by thecoder-co** [codemagic-ci-cd/codemagic-docs#1602] [Q&A] [answered]
Please i have a problem with building for ios.

```
💪 Building with sound null safety 💪

Archiving com.<ORG_NAME>.<APP_NAME>...
Automatically signing iOS for device deployment using specified development team in Xcode project: <PROJECT_ID>
Running pod install...                                             29.3s
Upgrading Pods-Runner-frameworks.sh
Running Xcode build...                                          
Xcode archive done.                                          4.3s
Failed to build iOS app
Error (Xcode): No profiles for 'com.<ORG_NAME>.<APP_NAME>' were found: Xcode couldn't find any iOS App Development provisioning profiles matching 'com.<ORG_NAME>.<APP_NAME>'. Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild.
/Users/builder/clone/ios/Runner.xcodeproj


It appears that there was a problem signing your application prior to installation on the device.

Verify that the Bundle Identifier in your project is your signing id in Xcode
  open ios/Runner.xcworkspace

Also try selecting 'Product > Build' to fix the problem.
Encountered error while archiving for device.


Build failed :|
Step 6 script `Flutter build ipa and automatic versioning` exited with status code 1
```

and i have even more than one profile for that identifier in the developer console
what else may be wrong?

**Comment by himesh-cm** [comment]
Hi @thecoder-co, Can you please share the build id? You can find it on the build details page.

**Reply by thecoder-co** [reply]
644b7cebee0c4fb84c106579 this is it, thanks so much for responding

**Reply by himesh-cm** [reply, marked as answer]
Hi @thecoder-co, 
1. You are missing the codesigning step from the workflow as shown [here](https://github.com/codemagic-ci-cd/codemagic-sample-projects/blob/main/flutter/flutter-android-and-ios-yaml-demo-project/codemagic.yaml#L55).
2. The profile you uploaded should also have the distribution certificate, as shown below. Currently, you have a development certificate.

<img width="1346" alt="Screenshot 2023-01-31 at 2 39 07 PM" src="https://user-images.githubusercontent.com/106069365/236187156-04a6cec5-cf52-4d7f-ba07-267e49107358.png">

**Reply by thecoder-co** [reply]
Thanks so much!
